File: cucumber-expressions/ruby/lib/cucumber/cucumber_expressions/undefined_parameter_type_expression.rb
require 'cucumber/cucumber_expressions/argument'
require 'cucumber/cucumber_expressions/tree_regexp'
require 'cucumber/cucumber_expressions/errors'

module Cucumber
  module CucumberExpressions
    class UndefinedParameterTypeExpression
      attr_reader :source

      def initialize(source)
        @source = source
      end

      def match(text)
        nil
      end

      def regexp
        /.^/
      end
    end
  end
end

File: cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/UndefinedParameterTypeExpression.java
package io.cucumber.cucumberexpressions;

import java.lang.reflect.Type;
import java.util.List;
import java.util.Set;
import java.util.regex.Pattern;

import static java.util.Collections.unmodifiableSet;

final class UndefinedParameterTypeExpression implements Expression {
    private final String source;
    private final Set<String> undefinedParameterTypeNames;

    UndefinedParameterTypeExpression(String source, Set<String> undefinedParameterTypeNames) {
        this.source = source;
        this.undefinedParameterTypeNames = undefinedParameterTypeNames;
    }

    @Override
    public List<Argument<?>> match(String text, Type... typeHints) {
        return null;
    }

    @Override
    public Pattern getRegexp() {
        return Pattern.compile(".^"); // matches nothing
    }

    @Override
    public String getSource() {
        return source;
    }

    @Override
    public Set<String> getUndefinedParameterTypeNames() {
        return unmodifiableSet(undefinedParameterTypeNames);
    }
}
